test/y2partitioner/dialogs/lvm_lv_info_test.rb

```diff
@@ -0,0 +1,179 @@
+    end
+
+    let(:value) { nil }
+
+    include_examples "CWM::AbstractWidget"
+
+    describe "#validate" do
+
+      context "when no lv name is entered" do
+        let(:value) { nil }
+
+        it "shows an error popup" do
+          expect(Yast::Popup).to receive(:Error)
+          widget.validate
+        end
+
+        it "return false" do
+          expect(widget.validate).to be(false)
+        end
+      end
+
+      context "when the entered lv name is not valid" do
+        let(:value) { "$!" }
+
+        it "shows an error popup" do
+          expect(Yast::Popup).to receive(:Error)
+          widget.validate
+        end
+
+        it "return false" do
+          expect(widget.validate).to be(false)
+        end
+      end
+
+      context "when the entered lv name is already used" do
+        let(:value) { "lv1" }
+
+        it "shows an error popup" do
+          expect(Yast::Popup).to receive(:Error)
+          widget.validate
+        end
+
+        it "return false" do
+          expect(widget.validate).to be(false)
+        end
+      end
+
+      context "when the entered lv name is valid" do
+        let(:value) { "lv3" }
+
+        it "does not show an error popup" do
+          expect(Yast::Popup).to_not receive(:Error)
+          widget.validate
+        end
+
+        it "return true" do
+          expect(widget.validate).to be(true)
+        end
+      end
+    end
+
+    describe "#store" do
+      let(:value) { "lv3" }
+
+      it "sets #lv_name in the controller" do
+        expect(controller.lv_name).to be_nil
+        widget.store
+        expect(controller.lv_name).to eq(value)
+      end
+    end
+  end
+
+  describe Y2Partitioner::Dialogs::LvmLvInfo::TypeWidget do
+    subject(:widget) { described_class.new(controller) }
+
+    before do
+      allow(widget).to receive(:value).and_return(value)
+    end
+
+    let(:value) { nil }
+
+    include_examples "CWM::AbstractWidget"
+
+    describe "#store" do
+      before do
+        allow(widget).to receive(:current_widget).and_return(current_widget)
+        allow(current_widget).to receive(:value).and_return(used_pool)
+      end
+
+      let(:current_widget) { Y2Partitioner::Dialogs::LvmLvInfo::ThinPoolSelector.new(controller) }
+
+      let(:used_pool) { nil }
+
+      let(:value) { :normal }
+
+      it "sets #type_choice in the controller" do
+        expect(controller.type_choice).to be_nil
+        widget.store
+        expect(controller.type_choice).to eq(value)
+      end
+
+      context "when selected type is :thin" do
+        let(:value) { :thin }
+
+        let(:used_pool) { "lv_thin_pool" }
+
+        it "sets #thin_pool in the controller" do
+          expect(controller.thin_pool).to be_nil
+          widget.store
+          expect(controller.thin_pool).to eq(used_pool)
+        end
+      end
+
+      context "when the selected type is not :thin" do
+        let(:value) { :normal }
+
+        it "does not set #thin_pool in the controller" do
```

Comment:
What if `controller.thin_pool` already contained a value? Is also set to `nil`?

Reply:
I isn't, and that is wrong. Fixing.